plugin.py

```diff
@@ -549,6 +564,21 @@ def unload(self):
         except:  # qgis 2.4
             self.legend.clicked.disconnect(self.on_legend_click)
 
+    def merge(self):
+        """merge branch into trunk"""
+        layer = QgsMapLayerRegistry.instance().mapLayer(
+            self.current_layers[0])
+        uri = QgsDataSourceURI(layer.source())
+        mtch = re.match(r'(.+)_([^_]+)_rev_(head|\d+)', uri.schema())
+        schema = mtch.group(1)
+        base_branch = mtch.group(2)
+        base_rev = mtch.group(3)
+
+        ret = versioning.merge(self.pg_conn_info(), schema, base_branch)
+
+        QMessageBox.information(self.iface.mainWindow(), "Merge",
+                                "%d feature(s) merged." % ret)
```

Comment:
Normally merging can cause conflicts if a feature has been edited on both sides (in the trunk and in the branch), I don't see this can happen here.

There may also be an option to delete a branch... but maybe not necessary.